src/Induction/WellFounded.agda

```diff
@@ -79,6 +89,26 @@ module All {a lt} {A : Set a} {_<_ : Rel A lt}
   Please use wfRecBuilder instead."
   #-}
 
+module FixPoint
+  {a ℓ} {A : Set a} {_<_ : Rel A ℓ} (wf : WellFounded _<_)
+  (P : Pred A ℓ) (f : WfRec _<_ P ⊆′ P)
+  (f-ext : (x : A) {IH IH' : WfRec _<_ P x} → (∀ y y<x → IH y y<x ≡ IH' y y<x) → f x IH ≡ f x IH')
+  where
+
+  someWfRecInv : ∀ x → (q q' : Acc _<_ x) → Some.wfRec P f x q ≡ Some.wfRec P f x q'
+  someWfRecInv = All.wfRec wf (a ⊔ ℓ)
+                           (λ x → (q q' : Acc _<_ x) → Some.wfRec P f x q ≡ Some.wfRec P f x q')
+                           (λ { x IH (acc rs) (acc rs') → f-ext x λ y y<x → IH y y<x (rs y y<x) (rs' y y<x) })
+
+  open All wf ℓ
+  wfRecFixpLemma : ∀ {x} y y<x → wfRecBuilder P f x y y<x ≡ wfRec P f y
+  wfRecFixpLemma {x} y y<x with wf x
+  wfRecFixpLemma {x} y y<x | acc rs = someWfRecInv y (rs y y<x) (wf y)
```

Comment:
I believe I cannot get rid of `with` because I want the lhs of the goal to reduce to `f y ...`, which will only happen after pattern matching on `wf x`.